lib/storage/file.js

```diff
@@ -104,8 +104,36 @@ var STORAGE_UPLOAD_BASE_URL = 'https://www.googleapis.com/upload/storage/v1/b';
  * var file = myBucket.file('my-file');
  */
 function File(bucket, name, options) {
+  options = options || {};
+
+  var requestQueryObject = {};
+  var generation = parseInt(options.generation, 10);
+
+  if (generation) {
```

Comment:
Relevant: https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/isNaN#Description

I'll switch it to `is.number(options.generation)`